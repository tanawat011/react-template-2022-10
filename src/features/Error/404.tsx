import { useNavigate } from 'react-router-dom'

import { Button } from 'components/Button'

export const Error404 = () => {
  const navigate = useNavigate()

  return (
    <div>
      <p>404 Not Found</p>

      <Button onClick={() => navigate(-1)}>Back to Previous Page</Button>
    </div>
  )
}
