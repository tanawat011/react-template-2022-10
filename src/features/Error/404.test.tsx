import { render } from '@testing-library/react'

import { Error404 } from './404'

describe('<Error404 />', () => {
  test('match snapshot', () => {
    const { asFragment } = render(<Error404 />)
    expect(asFragment()).toMatchSnapshot()
  })
})
