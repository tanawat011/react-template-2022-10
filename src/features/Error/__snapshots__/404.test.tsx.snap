// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error404 /> match snapshot 1`] = `
<DocumentFragment>
  <div>
    404 Not Found
  </div>
</DocumentFragment>
`;
